Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1126,41 +1126,17 @@ bool CLuaEngineDefs::EngineSetModelVisibleTime(const std::variant<std::string, u
     return modelInfo->SetTime(hourOn, hourOff);
 }
 
-int CLuaEngineDefs::EngineGetModelVisibleTime(lua_State* luaVM)
+std::tuple<char, char> CLuaEngineDefs::EngineGetModelVisibleTime(const std::variant<std::string, unsigned int> variantModelID)
 {
-    // int, int engineGetModelVisibleTime ( int/string modelID )
-    SString strModelId;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strModelId);
-
-    if (!argStream.HasErrors())
-    {
-        ushort      usModelID = CModelNames::ResolveModelID(strModelId);
-        CModelInfo* pModelInfo = g_pGame->GetModelInfo(usModelID);
-        if (pModelInfo)
-        {
-            char cHourOn, cHourOff;
-            if (pModelInfo->GetTime(cHourOn, cHourOff))
-            {
-                lua_pushnumber(luaVM, cHourOn);
-                lua_pushnumber(luaVM, cHourOff);
-                return 2;
-            }
-            else // Model is incompatible, don't let confuse user.
-            {
-                lua_pushnumber(luaVM, 0);
-                lua_pushnumber(luaVM, 24);
-                return 2;
-            }
-        }
-    }
-    else
-        luaL_error(luaVM, argStream.GetFullErrorMessage());
+    // either convert string to modelid using CModelNames::ResolveModelID, or assign the id from the variant
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
```

Comment:
EngineGetModelVisibleTime

```suggestion
    unsigned short modelID;
    if (std::holds_alternative<std::string>(variantModelID))
        modelID = CModelNames::ResolveModelID(std::get<std::string>(variantModelID));
    else
        modelID = std::get<unsigned int>(variantModelID);

```

Reply:
I'll just overload the `CModelNames::ResolveModelID` function to take an `variant` as an argument.